dali/kernels/imgproc/roi.h

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_ROI_H_
+#define DALI_KERNELS_IMGPROC_ROI_H_
+
+#include <vector>
+#include "dali/core/geom/box.h"
+#include "dali/kernels/tensor_shape.h"
+
+namespace dali {
+namespace kernels {
+
+/**
+ * Defines region of interest.
+ * 0 dimension is interpreted along x axis (horizontal)
+ * 1 dimension is interpreted along y axis (vertical)
+ *
+ *            image.x ->
+ *          +--------------------------------+
+ *          |                                |
+ *          |   roi.lo    roi.x              |
+ *  image.y |         +-----+                |
+ *       |  |    roi.y|     |                |
+ *       v  |         +-----+ roi.hi         |
+ *          |                                |
+ *          +--------------------------------+
+ *
+ * Additionally, by definition, ROI is top-left inclusive and bottom-right exclusive.
+ * That means, that `Roi.lo` point is included in actual ROI and `Roi.hi` point isn't.
+ */
+template <size_t ndims>
+using Roi = Box<ndims, int>;
+
+namespace detail {
+
+/**
+ * Create Roi for whole image defined by TensorShape
+ */
+template <size_t spatial_dims>
+Roi<spatial_dims> whole_image(const TensorShape<spatial_dims + 1> &shape) {
```

Comment:
```suggestion
Roi<spatial_dims> WholeImage(const TensorShape<spatial_dims + 1> &shape) {
```
to match naming convention in this file

Reply:
Done